lib/DxcBindingTable/DxcBindingTable.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+
+    if (integer.getBitWidth() > 32) {
+      return IntegerConversionStatus::OutOfBounds;
+    }
+
+    *outInteger = (uint32_t)integer.getLimitedValue();
+    return IntegerConversionStatus::Success;
+  }
+}
+
+bool hlsl::ParseBindingTable(llvm::StringRef fileName, llvm::StringRef content, llvm::raw_ostream &errors, DxcBindingTable *outTable) {
+
+  struct Parser {
+    StringRef fileName;
+    const char *curr = nullptr;
+    const char *end  = nullptr;
+    int line = 1;
+    int col  = 1;
+    llvm::raw_ostream &errors;
+    bool WasNewline = false;
+
+    struct Location {
+      int line = 0;
+      int col = 0;
+    };
+
+    inline static bool IsDelimiter(char c) {
+      return c == ',' || c == '\n' || c == '\r';
+    }
+    inline static bool IsNewline(char c) {
+       return c == '\r' || c == '\n';
+    }
+    inline static bool IsWhitespace(char c) {
+      return c == ' ' || c == '\t';
+    }
+    inline Parser(StringRef fileName, StringRef content, llvm::raw_ostream &errors) :
+      fileName(fileName),
+      curr(content.data()),
+      end(content.data() + content.size()),
+      errors(errors)
+    {
+      EatWhiteSpaceAndNewlines();
+    }
+    inline bool WasJustEndOfLine() const {
+      return WasNewline;
+    }
+
+    inline void EatWhitespace() {
+      for (;;) {
+        if (IsWhitespace(Peek()))
+          Advance();
+        else
+          break;
+      }
+    }
+    inline void EatWhiteSpaceAndNewlines() {
+      for (;;) {
+        if (IsWhitespace(Peek()) || IsNewline(Peek()))
+          Advance();
+        else
+          break;
+      }
+    }
+    inline Location GetLoc() const {
+      Location loc;
+      loc.line = line;
+      loc.col = col;
+      return loc;
+    }
+    void Advance() {
+      if (ReachedEnd())
+        return;
+      if (*curr == '\n') {
+        line++;
+        col = 1;
+      }
+      else if (*curr != '\r') {
+        col++;
+      }
+      curr++;
+    }
+    inline bool ReachedEnd() const {
+      return curr >= end || *curr == '\0';
+    }
+    inline void Warn(Location loc, const Twine &err) {
+      (void)Error(loc, err);
+    }
+    inline bool Error(Location loc, const Twine &err) {
+      errors << (Twine(fileName) + ":" + Twine(loc.line) + ":" + Twine(loc.col) + ": " + err + "\n").str();
+      return false;
+    }
+    inline bool Error(const Twine &err) {
+      Error(GetLoc(), err);
+      return false;
+    }
+    inline char Peek() const {
+      if (ReachedEnd()) return '\0';
+      return *curr;
+    }
+
+    bool ParseCell(SmallVectorImpl<char> *str) {
+      EatWhitespace();
+
+      if (ReachedEnd()) {
+        return Error("Unexpected EOF when parsing cell.");
+      }
+
+      bool hasQuote = false;
+      if (Peek() == '"') {
+        hasQuote = true;
+        Advance();
+      }
+
+      while (!ReachedEnd()) {
+        if (IsDelimiter(Peek())) {
+          if (hasQuote)
+            return Error("Expected end quote '\"'.");
```

Comment:
So you can't have a comma inside the quotes?  I feel like only newline should cause an error when inside quotes.

Reply:
That's a mistake. It's been fixed now and tested.